src/TLuaInterpreter.cpp

```diff
@@ -4243,6 +4259,80 @@ int TLuaInterpreter::setLabelOnLeave(lua_State* L)
     return setLabelCallback(L, qsl("setLabelOnLeave"));
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setMovie
+int TLuaInterpreter::setMovie(lua_State* L)
+{
+    QString labelName = getVerifiedString(L, __func__, 1, "label name");
+    if (labelName.isEmpty()) {
+        return warnArgumentValue(L, __func__, "label name cannot be an empty string");
+    }
+    QString moviePath = getVerifiedString(L, __func__, 2, "movie (gif) path");
+
+    Host& host = getHostFromLua(L);
+    if (auto [success, message] = host.setMovie(labelName, moviePath); !success) {
+        return warnArgumentValue(L, __func__, message);
+    }
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// No documentation available in wiki - internal function
+int TLuaInterpreter::movieFunc(lua_State* L, const QString& funcName)
+{
+    QString labelName = getVerifiedString(L, funcName.toUtf8().constData(), 1, "label name");
+    if (labelName.isEmpty()) {
+        return warnArgumentValue(L, __func__, "label name cannot be an empty string");
+    }
+    auto pN = LABEL(L, labelName);
+    lua_remove(L, 1);
+    auto movie = pN->movie();
+    if (!movie) {
+        return warnArgumentValue(L, __func__, qsl("no movie found at label '%1'").arg(labelName));
+    }
+
+    if (funcName == qsl("setMovieStart")) {
+        movie->start();
+    } else if (funcName == qsl("setMoviePaused")) {
+        movie->setPaused(movie->state() != QMovie::Paused);
+    } else if (funcName == qsl("setMovieFrame")) {
+        int frame = getVerifiedInt(L, funcName.toUtf8().constData(), 1, "movie frame number");
+        lua_pushboolean(L, movie->jumpToFrame(frame));
+        return 1;
+    } else if (funcName == qsl("setMovieSpeed")) {
+        int speed = getVerifiedInt(L, funcName.toUtf8().constData(), 1, "movie playback speed in %");
```

Comment:
Didn't have a chance to test in action, but reading code and documentation above, these should be argument #2 not #1 right?
```suggestion
        int frame = getVerifiedInt(L, funcName.toUtf8().constData(), 2, "movie frame number");
        lua_pushboolean(L, movie->jumpToFrame(frame));
        return 1;
    } else if (funcName == qsl("setMovieSpeed")) {
        int speed = getVerifiedInt(L, funcName.toUtf8().constData(), 2, "movie playback speed in %");
```

Reply:
I think this isn't needed as the first argument is removed at line 4287 by using lua_remove(L, 1);

Reply:
You're probably right, and it just confused me profoundly, as hardly any other functions in TLuaInterpreter use this lua_remove() functionality. Instead they all verify their arguments using the same count as documented and given by the player. Would be nice to use consistent style throughout.

Reply:
Agreed, this lua_remove is kinda sneaky and throws you off.

Reply:
I got it by copying from setLabelCallback will get rid of it